api/src/main/java/jakarta/data/metamodel/restrict/TextRestrictionRecord.java

```diff
@@ -17,76 +17,46 @@
  */
 package jakarta.data.metamodel.restrict;
 
-// Internal implementation class.
-// The proper way for users to obtain instances is via
-// the static metamodel or Restrict.* methods 
+import jakarta.data.metamodel.constraint.Like;
 
 import java.util.Objects;
 
-record TextRestrictionRecord<T>(
-        String attribute,
-        Operator comparison,
-        boolean isCaseSensitive,
-        boolean isEscaped,
-        String value) implements TextRestriction<T> {
+record TextRestrictionRecord<T>(String attribute, Like constraint, boolean negated)
+        implements TextRestriction<T> {
 
     TextRestrictionRecord {
         Objects.requireNonNull(attribute, "Attribute must not be null");
+        Objects.requireNonNull(constraint, "Pattern must not be null");
```

Comment:
Made more generic in case it is a Literal rather than a Pattern
```suggestion
        Objects.requireNonNull(constraint, "Constraint must not be null");
```